multimap/src/main/java/org/infinispan/multimap/impl/Bucket.java

```diff
@@ -138,8 +149,28 @@ public List<V> toList() {
       return new ArrayList<>(values);
    }
 
+   /**
+    * @return a defensive copy of the {@link #values} collection if the cache supports duplicates.
+    */
+   public List<V> toLinkedList() {
+      return new LinkedList<>(values);
```

Comment:
Do we have a risk of changing the order here? Just wondering, since the collection implementation, when serializing, it is an ArrayList. When we move values between nodes if the ordering could get messed up.